### PR TITLE
Game stats

### PR DESCRIPTION
**Pull request by jpc20**
Add methods for game statistics.
All tests are passing but it will need to be refactored.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -11,10 +14,83 @@ def self.from_csv(locations)
...
+    games_by_season.transform_values {|season| season.length}
+  end
+  def highest_total_score
+    highest_scoring_game = @games.max_by do |game| game.away_goals + game.home_goals
+    end
+    highest_scoring_game.away_goals + highest_scoring_game.home_goals
+  end
+
+  def lowest_total_score
+    lowest_scoring_game = @games.min_by do |game| game.away_goals + game.home_goals
```
Comment (jpc20) [inline lib/stat_tracker.rb:56]: New line.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -11,10 +14,83 @@ def self.from_csv(locations)
...
+    ties = @game_teams.find_all {|gmteam| gmteam.result == "TIE"}
+    ((ties.length.to_f / @game_teams.length) * 100).round(2)
+  end
+
+  def count_of_games_by_season
+    games_by_season = @games.group_by {|game| game.season}
+    games_by_season.transform_values {|season| season.length}
+  end
+  def highest_total_score
+    highest_scoring_game = @games.max_by do |game| game.away_goals + game.home_goals
```
Comment (jpc20) [inline lib/stat_tracker.rb:50]: New line.

File: lib/game.rb [hunk 3]
```diff
@@ -24,7 +24,6 @@ def self.all
   end
 
   def initialize(game_info)
-    @game_info = game_info
```
Comment (arpariseau) [inline lib/game.rb:27]: we didn't need to save the game info, since we're parsing it into all the other variables

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -11,10 +14,83 @@ def self.from_csv(locations)
...
+  def percentage_away_wins
+    away_wins = @games.find_all do |game|
+      game.away_goals > game.home_goals
+    end
+    ((away_wins.length.to_f / @games.length.to_f) * 100).round(2)
+  end
+
+  def percentage_ties
+    ties = @game_teams.find_all {|gmteam| gmteam.result == "TIE"}
+    ((ties.length.to_f / @game_teams.length) * 100).round(2)
```
Comment (jpc20) [inline lib/stat_tracker.rb:42]: Let's add percentage to a module.

File: lib/stat_tracker.rb [hunk 5]
```diff
@@ -1,7 +1,10 @@
 require 'csv'
+require_relative 'game_team'
+require_relative 'game'
+require_relative 'team'
 
 class StatTracker
-  attr_reader :games_path, :teams_path, :game_teams_path
```
Comment (arpariseau) [inline lib/stat_tracker.rb:4]: Didn't need to hang onto the path, much more important to hold onto the games, teams and game_teams (seriously, there needs to be a better term for this)

File: test/game_test.rb [hunk 6]
```diff
@@ -38,7 +38,7 @@ def test_it_has_attributes
     assert_equal   '/api/v1/venues/null' , @game.venue_link
   end
 
-  def test_it_can_read_games_from_CSV
+  def def_test_it_can_read_games_from_CSV
```
Comment (arpariseau) [inline test/game_test.rb]: I dunno how that got screwed back up, but we need to fix it again

File: test/stat_tracker_test.rb [hunk 7]
```diff
@@ -16,9 +17,64 @@ def test_it_exists
...
+  def test_percentage_home_wins
+    assert_equal 63.64, @stat_tracker.percentage_home_wins
+  end
+
+  def test_percentage_home_wins
+    assert_equal 27.27, @stat_tracker.percentage_away_wins
+  end
+
+  def test_percentage_ties
+    assert_equal ((1.0 / 11) * 100).round(2), @stat_tracker.percentage_ties
```
Comment (arpariseau) [inline test/stat_tracker_test.rb:34]: Switch this from a calculation to just the number, 9.09.

File: lib/stat_tracker.rb [hunk 8]
```diff
@@ -11,10 +14,83 @@ def self.from_csv(locations)
...
+
+  def sum_of_goals_in_a_season(season)
+    full_season = by_season(season)
+    full_season.sum do |game|
+      game.home_goals + game.away_goals
+    end
+  end
+
+  def average_of_goals_in_a_season(season)
+    (sum_of_goals_in_a_season(season).to_f / by_season(season).length).round(2)
```
Comment (jpc20) [inline lib/stat_tracker.rb:85]: We could also add average to a module

**Pull request merged**